docs/QUICKSTART.md

```diff
@@ -0,0 +1,410 @@
+* An IRQ that belongs to a box can only be modified when that box context is active.
+
+Although this behaviour is different from the original NVIC one, it is backwards compatible. This means that legacy code (like a device HAL) will still work after uVisor is enabled. The general use case is the following:
+
+```C
+#define MY_IRQ 42
+
+/* Set the ISR for MY_IRQ at runtime.
+ * Without uVisor: Relocate the vector table to SRAM and set my_isr as the ISR for MY_IRQ.
+ * With    uVisor: Register MY_IRQ for the current box with my_isr as ISR. */
+NVIC_SetVector(MY_IRQ, &my_isr);
+
+/* Change the IRQ state. */
+NVIC_SetPriority(MY_IRQ, 3);
+NVIC_EnableIRQ(MY_IRQ);
+```
+
+> **Note**: In this model a call to `NVIC_SetVector` must always happen before an IRQ state is changed. In platforms that don't relocate the vector table such call might be originally absent and must be added to work with uVisor.
+
+For more information on the uVisor APIs, checkout the [uVisor APIs documentation](APIS.md).
+
+### The *main box* ACLs
+
+The code samples that we provide in this guide give you a ready-made list of ACLs for the main box. The list includes peripherals that we already know will be necessary to make the example app work, and it is specific to the NXP FRDM-K64F target.
+
+This section shows how to discover the needed ACLs for the main box. You might need to follow these instructions in case you want to generate the ACLs list for a different target or a different app.
+
+At the moment the uVisor does not provide a way to detect and list all the faulting ACLs for a given platform automatically. This is a planned feature that will be released in the future.
+
+In order to generate the list of ACLs, use the code provided in the [Enable uVisor](#enable-uvisor) section. In this case, though, start with an empty ACLs list:
+
+```C
+static const UvisorBoxAclItem g_main_box_acls[] = {
+}
+```
+
+You now need to compile your application using uVisor in debug mode. This operation requires some more advanced steps, which are described in detail in the [Debugging uVisor](DEBUGGING.md) document. The main idea is that you compile the application in debug mode:
+
+```bash
+$ mbed compile -m K64F_SECURE -t GCC_ARM -o "debug-info"
+```
+
+and then use a GDB-compatible interface to flash the device, enable semihosting, and access the uVisor debug messages. Please read the [Debugging uVisor](DEBUGGING.md) document for the detailed instructions.
+
+Once the uVisor debug messages are enabled, you will see you application fail. The failure is due to the first missing ACL being hit by the main box code. The message will look like:
+
+```
+***********************************************************
+                    BUS FAULT
+***********************************************************
+
+...
+
+* MEMORY MAP
+  Address:           0x4004800C
+  Region/Peripheral: SIM
```

Comment:
Sure but they have immediate access here to the "alternative" solution. It think the guide is nicer with the peripheral names 😄

...And when you don't see them on your screen and come back to the guide to complain, you'll see that there's a note addressing your problem already 😉